zk-token-sdk/src/instruction/range_proof.rs

```diff
@@ -0,0 +1,122 @@
+//! The range proof instruction.
+//!
+//! A range proof certifies that a committed value in a Pedersen commitment is a number from a
+//! certain range. Currently, only 64-bit range proof `VerifyRangeProof64` is supported in the
+//! proof program. It certifies that a committed number is an unsigned 64-bit number.
+
+#[cfg(not(target_os = "solana"))]
+use {
+    crate::{
+        encryption::pedersen::{PedersenCommitment, PedersenOpening},
+        errors::ProofError,
+        range_proof::RangeProof,
+        transcript::TranscriptProtocol,
+    },
+    merlin::Transcript,
+    std::convert::TryInto,
+};
+use {
+    crate::{
+        instruction::{ProofType, ZkProofData},
+        zk_token_elgamal::pod,
+    },
+    bytemuck::{Pod, Zeroable},
+};
+
+#[cfg(not(target_os = "solana"))]
+const RANGEPROOF64_BIT_LENGTH: usize = 64;
```

Comment:
Would it be possible to use u64::BITS instead of adding this new constant?

Reply:
Oh, I never knew/used `u64::BITS` before. I think it will work!